src/main/kotlin/org/jitsi/nlj/transform/node/outgoing/AbsSendTime.kt

```diff
@@ -25,19 +25,21 @@ import org.jitsi.nlj.util.cdebug
 import org.jitsi.rtp.rtp.header_extensions.AbsSendTimeHeaderExtension
 import org.jitsi.rtp.rtp.header_extensions.HeaderExtensionType
 import org.jitsi.rtp.NewRawPacket
+import org.jitsi.rtp.rtp.RtpPacket
 import unsigned.toUInt
 import java.nio.ByteBuffer
 
 class AbsSendTime : TransformerNode("Absolute send time") {
-    private var extensionId: Int = -1
+    private var extensionId: Int? = null
 
     override fun transform(packetInfo: PacketInfo): PacketInfo? {
-        val absSendTimeExt = AbsSendTimeHeaderExtension(extensionId, System.nanoTime());
-        val extData = ByteArray(absSendTimeExt.sizeBytesAs(HeaderExtensionType.ONE_BYTE_HEADER_EXT) - 1)
-        absSendTimeExt.serializeData(ByteBuffer.wrap(extData))
-//        absSendTimeExt.serializeToAs(HeaderExtensionType.ONE_BYTE_HEADER_EXT, ByteBuffer.wrap(extData))
-//        packetInfo.packetAs<RtpPacket>().header.addExtension(extensionId, absSendTimeExt)
-        packetInfo.packetAs<NewRawPacket>().addExtension(extensionId.toByte(), extData)
+        extensionId?.let { absSendTimeExtId ->
+            val rtpPacket = packetInfo.packetAs<RtpPacket>()
+            val ext = rtpPacket.getHeaderExtension(absSendTimeExtId.toByte()) ?:
```

Comment:
If we're going to be doing toByte() for every packet, should we store it as a byte in the first place? Same applies to TccSeqNumTagger

Reply:
yep, this is addressed in an upcoming set of changes